File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        list=arr.copy()
        list.sort()
        j=1
        d={}
        for i in range(0, len(list)):
            if list[i] not in d:
                d[list[i]]=j
                j+=1
        for i in range(0, len(arr)):
            arr[i]=d[arr[i]]
        return arr

File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int n=arr.length;
        int ans[]=Arrays.copyOf(arr,n);
        Arrays.sort(ans);
        HashMap<Integer, Integer> mp=new HashMap<>();
        int j=1;
        for(int i:ans){
            if(!mp.containsKey(i))mp.put(i,j++);
        }
        for(int i=0; i<n; i++){
            ans[i]=mp.get(arr[i]);
        }
        return ans;
    }
}